CHANGELOG.md

```diff
@@ -6,6 +6,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Refactor
+
+- Replaced old TDXApiError with new ES6 `extends Error` syntax.
+  This is not supported by Internet Explorer 10.
+- Adds `*.d.ts` type defintions for TypeScript/VS Code/other IDEs.
+
+### Dependencies
+
+- Bump semver of dependencies:
+  - @nqminds/nqm-core-utils: ^0.5.9
+  - base-64: ^1.0.0
+  - cross-fetch: ^3.1.2
+  - debug: ^4.3.1
+- Update devDepdencies to work with NPM v7 `npm install`:
```

Comment:
I assume you've tested this with NPM < 7 as well? Think most people don't upgrade _that_ often.

Reply:
NPM < 4 will not work, since `npm run prepare` only became a feature in version 4.0.0, but the last NPMv3 release was in 2016, so I think we're probably okay on that side.

NPM version 6 works, but prints out a warning that it can't fully understand the `package-lock.json` file (it's backwards compatible with NPMv5/v6). The only bad thing is that doing if people are constantly developing with NPM v6/v7, the lockfile will keep adding/deleting the extra `v2` data that NPMv7 injects. But it's would just make our `git diffs` look bigger, both lock-files are compatible with each version of NPM.

```
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
```